client/my-sites/stats/all-time-highlights-section/post-cards-group.tsx

```diff
@@ -60,20 +61,19 @@ const getStatsQueries = createSelector(
 	( state, siteId ) => getSiteOption( state, siteId, 'gmt_offset' )
 );
 
-const getStatsData = createSelector(
+const getStatsTopPostsData = createSelector(
 	// eslint-disable-next-line @typescript-eslint/no-unused-vars
-	( state, siteId, topPostsQuery, isTopViewedPostRequesting ) => {
+	( state, siteId, topPostsQuery, hasTopViewedPostQueryFinished ) => {
 		const { post: topPost, page: topPage } = getTopPostAndPage( state, siteId, topPostsQuery );
 
 		return {
 			topPost,
 			topPage,
 		};
 	},
-	// Refresh memoized data from createSelector with the loading flag
-	( state, siteId, topPostsQuery, isTopViewedPostRequesting ) => [
+	( state, siteId, topPostsQuery ) => [
 		topPostsQuery,
-		isTopViewedPostRequesting,
+		getSiteOption( state, siteId, 'gmt_offset' ),
```

Comment:
Theoretically the timezone setting of a site wouldn't frequently change, any particular reason we need `gmt_offset` as cache key here?

Reply:
I just tried adding another cache key to ensure the memorization is correct. Indeed, The `gmt_offset` has affected the `topPostsQuery` already.